### ISSUE TITLE
how does Keras handle loss in a batch?

### ISSUE THREAD

**Issue by camigord** [AppliedDataSciencePartners/WorldModels#4]
Hi, thanks for the implementation!

I have a question related to how Keras handles the loss in a batch. I noticed that the loss in the RNN is defined as:

```
result = K.mean(result, axis = (1,2)) # mean over rollout length and z dim
```

That means that you end up with one loss per sample in the batch. I am not very familiar with Keras, so I wonder about what happens with these losses when you call 

```
rnn.fit(rnn_input, rnn_output,
        shuffle=True,
        epochs=mdn_epochs,
        batch_size=batch_size,
        validation_split=0.2,
        callbacks=callbacks_list)
```

Does Keras average over the batch? Could you point me to some documentation about it?

Thanks a lot!

**Comment by davidADSP** [comment]
Yes that's correct I believe:
see here https://keras.io/losses/
"The actual optimized objective is the mean of the output array across all datapoints."
